examples/bond/test/common/common.js

```diff
@@ -0,0 +1,254 @@
+};
+
+function optInLsigToBond (runtime, lsig, assetID, appManager) {
+  // Only store manager can allow opt-in to ASA for lsig
+  const optInTx = [
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: appManager.account,
+      toAccountAddr: lsig.address(),
+      amountMicroAlgos: 0,
+      payFlags: {}
+    },
+    {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: lsig.address(),
+      lsig: lsig,
+      assetID: assetID,
+      payFlags: {}
+    }
+  ];
+  runtime.executeTx(optInTx);
+};
+
+const placeholderParam = {
+  TMPL_NOMINAL_PRICE: 1000,
+  TMPL_MATURITY_DATE: Math.round(new Date().getTime() / 1000) + 240
+};
+const approvalProgram = getProgram('bond-dapp-stateful.py', placeholderParam);
+const clearProgram = getProgram('bond-dapp-clear.py');
+
+const minBalance = 10e6; // 10 ALGO's
+const initialBalance = 200e6;
+const coupon = 20;
+const issue = 1000;
+
+/**
+ * Creates DEX_i lsig, burn B_i tokens, issue B_i+1 tokens
+ * @param runtime runtime object
+ * @param creatorAccount
+ * @param managerAcc
+ * @param i create ith dex
+ * @param master
+ * @param issuerLsig
+ * i must be >= 1
+ */
+function createDex (runtime, creatorAccount, managerAcc, i, master, issuerLsig) {
+  if (i < 1) {
+    throw new Error('i must be greater than equal to 1');
+  }
+
+  const previousToken = bondToken + String(i - 1);
+  const oldBond = runtime.getAssetInfoFromName(previousToken).assetIndex;
+  const appInfo = runtime.getAppInfoFromName(approvalProgram, clearProgram);
+  const newBondToken = bondToken + String(i);
+  const getGlobal = (key) => runtime.getGlobalState(appInfo.appID, key);
+
+  // Create B_[i+1]
+  const newBond = runtime.addASADef(
+    newBondToken,
+    asaDef,
+    { creator: { ...creatorAccount.account, name: 'bond-token-creator' } }
+  );
+
+  optInLsigToBond(runtime, issuerLsig, newBond, managerAcc);
+
+  // Create dex
+  const param = {
+    TMPL_OLD_BOND: oldBond,
+    TMPL_NEW_BOND: newBond,
+    TMPL_APPLICATION_ID: appInfo.appID,
+    TMPL_APP_MANAGER: managerAcc.address
+  };
+  const dexLsigProgram = getProgram('dex-lsig.py', param);
+  const dexLsig = runtime.getLogicSig(dexLsigProgram, []);
+  const dexLsigAddress = dexLsig.address();
+
+  // fund dex with some minimum balance first
+  const fundDexParam = {
+    type: types.TransactionType.TransferAlgo,
+    sign: types.SignType.SecretKey,
+    fromAccount: master.account,
+    toAccountAddr: dexLsigAddress,
+    amountMicroAlgos: minBalance + 10000,
+    payFlags: {}
+  };
+  runtime.executeTx(fundDexParam);
+
+  optInLsigToBond(runtime, dexLsig, oldBond, managerAcc);
+  optInLsigToBond(runtime, dexLsig, newBond, managerAcc);
+
+  const total = getGlobal('total');
+  const assetAmount = runtime.getAccount(issuerLsig.address()).getAssetHolding(oldBond)?.amount;
+  const groupTx = [
```

Comment:
let's move this groupTx creation to a function in examples/bond/scripts/run/common/common.js. It will be useful for a user who will like to use it in scripts. 
Also - in our example script we have to create a DEX - so we should reuse it.

Reply:
ok added a function with 10 parameters.